pkg/tenant/projects.go

```diff
@@ -514,3 +514,67 @@ func (s *Server) createProject(ctx context.Context, project *db.Project) (err er
 
 	return nil
 }
+
+// UpdateProjectStats updates the stat fields on a project by performing readonly
+// queries to trtl and Quarterdeck. Because this requires a few RPCs, it should be
+// called in a background task where possible to avoid blocking user requests. The
+// context passed to this method must contain authentication credentials in order to
+// query Quarterdeck.
+// TODO: This data can be updated asynchronously once the Ensign "meta" topics are up
+// and running.
+func (s *Server) UpdateProjectStats(ctx context.Context, projectID ulid.ULID) (err error) {
+	var numTopics, numAPIKeys uint64
+
+	// Count the number of keys in Quarterdeck.
+	req := &qd.APIPageQuery{
+		ProjectID: projectID.String(),
+		PageSize:  100,
+	}
+
+	// Count the number of API keys in Quarterdeck.
+	for {
+		var reply *qd.APIKeyList
+		if reply, err = s.quarterdeck.APIKeyList(ctx, req); err != nil {
+			return err
+		}
+
+		numAPIKeys += uint64(len(reply.APIKeys))
+		if reply.NextPageToken == "" {
+			break
+		}
+
+		req.NextPageToken = reply.NextPageToken
+	}
```

Comment:
Yeh, I can see the issue; this isn't particularly performant -- probably what we should do is have a Quarterdeck endpoint that returns the number of API keys -- I'll create a story for that.